### ISSUE TITLE
Error ketika menambah layer peta bencana

### ISSUE THREAD

**Issue by anaritawidya** [inasafe/inasafe#4738]
### Problem
Phyton error ketika menambah layer peta bencana
![phyton error](https://user-images.githubusercontent.com/34257163/33595788-06ffbdc0-d9cc-11e7-9cae-a2b1c79bcac0.JPG)

```

2017-12-05T14:49:02	1	Traceback (most recent call last):
			  File "C:/Users/HP/.qgis2/python/plugins\inasafe\safe\plugin.py", line 962, in layer_changed
			    layer_purpose, layer_subcategory)
			  File "C:/Users/HP/.qgis2/python/plugins\inasafe\safe\definitions\utilities.py", line 461, in get_field_groups
			    field_groups += deepcopy(subcategory['field_groups'])
			KeyError: 'field_groups'
			
2017-12-05T14:49:53	1	Traceback (most recent call last):
			  File "C:/Users/HP/.qgis2/python/plugins\inasafe\safe\plugin.py", line 962, in layer_changed
			    layer_purpose, layer_subcategory)
			  File "C:/Users/HP/.qgis2/python/plugins\inasafe\safe\definitions\utilities.py", line 461, in get_field_groups
			    field_groups += deepcopy(subcategory['field_groups'])
			KeyError: 'field_groups'
			
2017-12-05T14:49:58	1	Traceback (most recent call last):
			  File "C:/Users/HP/.qgis2/python/plugins\inasafe\safe\plugin.py", line 962, in layer_changed
			    layer_purpose, layer_subcategory)
			  File "C:/Users/HP/.qgis2/python/plugins\inasafe\safe\definitions\utilities.py", line 461, in get_field_groups
			    field_groups += deepcopy(subcategory['field_groups'])
			KeyError: 'field_groups'
```

### Environment
* InaSAFE : 4.2.5
* QGIS : 2.18.14
* OS : win 10
* Data used (filenames and paths so as to reproduce, a project is better):
  * Exposure : 
  * Hazard : 
  * Aggregation :
  * Extent : 
  
Checklist (if applicable): 
* [ ] Did you try to run it in debug mode ?
* [ ] Is the data available ?
  * If yes, where (BTSync, data.inasafe.org, zip) ? 

### Proposed Solution

@ismailsunni @felix-yew

**Comment by ismailsunni** [comment]
How to replicate the issue:
1. Load layer with old keywords, that created by old InaSAFE (e.g. 3.5) with generic hazard
2. It will give you the same issue like @anaritawidya got

The reason:
1. InaSAFE used `generic` key for hazard generic.
2. We move to use `hazard_generic` in version 4.
3. InaSAFE try to resolve the `generic` key, but it couldn't, so return empty dictionary
4. InaSAFE try to access a key from the empty dict, error.

**Comment by ismailsunni** [comment]
FIxed in #4742
